graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransport.java

```diff
@@ -200,6 +237,28 @@ public ConfigurationRequest getRequestedConfiguration() {
                             TextField.Attribute.IS_PASSWORD
                     )
             );
+            x.addField(
+                    new BooleanField(
+                            CK_TLS_NEED_CLIENT_AUTH,
+                            "TLS Need Client Auth",
+                            false,
+                            "TLS Need Client Auth")
```

Comment:
Also setting "need client auth" does not have any effect. I can still send with a client that has no client certificates configured. Only "want client auth" seems to make a difference. If it is checked, the connection fails if no client certs are configured.

Can someone explain why we need both? @gbu-censhare